modules/service/src/main/scala/lucuma/odb/graphql/mapping/AccessControl.scala

```diff
@@ -330,5 +375,50 @@ trait AccessControl[F[_]] extends Predicates[F] {
             case Nil => Result(AccessControl.Checked.Empty)
             case List(x) => Services.asSuperUser(Result(AccessControl.unchecked(input, x, observation_id)))
             case _ => Result.internalError("Unpossible: got more than one oid back")
-    
+
+
+  def selectForUpdate(
+    input: CreateObservationInput,
+  )(using Services[F]): F[Result[AccessControl.CheckedWithId[ObservationPropertiesInput.Create, Program.Id]]] = {
+
+    val ensureWritable: F[Result[Option[Program.Id]]] =
+      programService
+        .resolvePid(input.programId, input.proposalReference, input.programReference)
+        .flatMap: r =>
+          r.flatTraverse: pid =>
+            selectForProgramUpdateImpl(None, List(pid))
+              .map: r =>
+                r.flatMap:
+                  case List(pid) => Result(Some(pid))
+                  case Nil       => Result(None)
+                  case _         => Result.internalError("Unpossible: selectForProgramUpdateImpl returned multiple ids")
+
+
+    // Compute our ObservationPropertiesInput.Create and set/verify the science band
+    def props(pid: Program.Id): F[Result[ObservationPropertiesInput.Create]] =
+      val props = input.SET.getOrElse(ObservationPropertiesInput.Create.Default)
+      props.scienceBand match
+        case None =>
+          allocationService
+            .selectScienceBands(pid)
+            .map(_.toList)
+            .map:
+              case List(b) => Result(props.copy(scienceBand = Some(b)))
+              case _       => Result(props)
```

Comment:
I wonder if we should pick the lowest band.  I guess we weren't doing that in the old code though.